security/tpm/devdoc/tpm.md

```diff
@@ -0,0 +1,60 @@
+# azure-iot-security-tpm.TpmSecurityClient Requirements
+
+## Overview
+`TpmSecurityClient` provides query and key operations for the device registration client.
+
+## Example usage
+TBD
+
+### TpmSecurityClient(registrationId?: string, customTpm?: any) [constructor]
+
+The `TpmSecurityClient` constructor initializes a new instance of a `TpmSecurityClient` object that is used to query TPM entities and perform key manipulation and setup.
+
+**SRS_NODE_TPM_SECURITY_CLIENT_06_001: [** The `registrationId` argument if present will be returned as the `registrationId` for subsequent calls to `getRegistrationId`. **]**
+
+**SRS_NODE_TPM_SECURITY_CLIENT_06_002: [** The `customTpm` argument, if present` will be used at the underlying TPM provider.  Otherwise the TPM provider will the tss TPM client with a parameter of `false` for simulator use. **]**
+
+### getEndorsementKey(callback: (err: Error, endorsementKey: string) => void): void
+
+Queries and returns the public portion of the endorsement key in the TPM hardware.
+
+**SRS_NODE_TPM_SECURITY_CLIENT_06_006: [** The `getEndorsmentKey` function shall query the TPM hardware and return the `endorsementKey` in the callback. **]**
```

Comment:
>query the TPM hardware [](start = 81, length = 22)

if at all possible considering how hairy the code might be, it'd be nice to have requirements telling what methods are being called and what is expected of them (so that tests know what to mock for the lower layer)